apm-sniffer/apm-sdk-plugin/finagle-6.25.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/finagle/ClientDestTracingFilterInterceptor.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.finagle;
+
+import com.twitter.finagle.Address;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+
+import java.lang.reflect.Method;
+import java.net.InetSocketAddress;
+
+import static org.apache.skywalking.apm.plugin.finagle.ContextCarrierHelper.setPeerHost;
+import static org.apache.skywalking.apm.plugin.finagle.FinagleCtxs.getContextCarrier;
+import static org.apache.skywalking.apm.plugin.finagle.FinagleCtxs.getSpan;
+
+public class ClientDestTracingFilterInterceptor extends AbstractInterceptor {
+
+    @Override
+    public void onConstructImpl(EnhancedInstance enhancedInstance, Object[] objects) {
+        enhancedInstance.setSkyWalkingDynamicField(getRemote(objects));
+    }
+
+    @Override
+    public void beforeMethodImpl(EnhancedInstance enhancedInstance, Method method, Object[] objects, Class<?>[] classes, MethodInterceptResult methodInterceptResult) throws Throwable {
+        String peer = (String) enhancedInstance.getSkyWalkingDynamicField();
+        AbstractSpan span = getSpan();
+        if (span != null) {
+            span.setPeer(peer);
+        }
+        SWContextCarrier swContextCarrier = getContextCarrier();
+        if (swContextCarrier != null) {
+            setPeerHost(swContextCarrier.carrier(), peer);
+        }
+    }
+
+    @Override
+    public Object afterMethodImpl(EnhancedInstance enhancedInstance, Method method, Object[] objects, Class<?>[] classes, Object o) throws Throwable {
+        return o;
+    }
+
+    @Override
+    public void handleMethodExceptionImpl(EnhancedInstance enhancedInstance, Method method, Object[] objects, Class<?>[] classes, Throwable throwable) {
+
```

Comment:
Exception information to be added like `ContextManager.activeSpan().errorOccurred().log(t)`

Reply:
Sure, I will add it.